components/brave_wallet/browser/asset_discovery_manager.cc

```diff
@@ -330,6 +337,274 @@ void AssetDiscoveryManager::MergeDiscoveredEthAssets(
+AssetDiscoveryManager::ParseNFTsFromSimpleHash(const base::Value& json_value,
+                                               mojom::CoinType coin) {
+  // Only ETH and SOL NFTs are supported.
+  if (!(coin == mojom::CoinType::ETH || coin == mojom::CoinType::SOL)) {
+    return absl::nullopt;
+  }
+
+  const base::Value::Dict* dict = json_value.GetIfDict();
+  if (!dict) {
+    return absl::nullopt;
+  }
+
+  GURL nextURL;
+  auto* next = dict->FindString("next");
+  if (next) {
+    nextURL = GURL(*next);
+  }
+
+  // Validate the URL of the next page using HTTPS and SimpleHash host
+  // and replace the host with the proxy host
+  if (!nextURL.is_empty() && (nextURL.host() != GURL(kSimpleHashUrl).host() ||
+                              !(nextURL.scheme() == url::kHttpsScheme))) {
+    nextURL = GURL();
+  } else {
+    GURL::Replacements replacements;
+    std::string replacement_host = GURL(kSimpleHashBraveProxyUrl).host();
+    replacements.SetHostStr(replacement_host);
+    nextURL = nextURL.ReplaceComponents(replacements);
+  }
+
+  const base::Value::List* nfts = dict->FindList("nfts");
+  if (!nfts) {
+    return absl::nullopt;
+  }
+
+  const base::flat_map<std::string, std::string>& chain_ids =
+      FromSimpleHashChainId();
+  std::vector<mojom::BlockchainTokenPtr> nft_tokens;
+  for (const auto& nft_value : *nfts) {
+    auto token = mojom::BlockchainToken::New();
+
+    // Skip all tokens with a collection.spam_score > 0.
+    const base::Value::Dict* nft = nft_value.GetIfDict();
+    auto* collection = nft->FindDict("collection");
+    if (!collection) {
+      continue;
+    }
+    absl::optional<int> spam_score = collection->FindInt("spam_score");
+    if (!spam_score || *spam_score > 0) {
+      continue;
+    }
+
+    // contract_address (required)
+    auto* contract_address = nft->FindString("contract_address");
+    if (!contract_address) {
+      continue;
+    }
+    token->contract_address = *contract_address;
+
+    // chain_id (required)
+    auto* chain = nft->FindString("chain");
+    if (!chain || !chain_ids.contains(*chain)) {
+      continue;
+    }
+    token->chain_id = chain_ids.at(*chain);
+
+    // name
+    auto* name = nft->FindString("name");
+    if (name) {
+      token->name = *name;
+    }
+
+    // logo
+    auto* logo = nft->FindString("image_url");
+    if (logo) {
+      token->logo = *logo;
+    }
+
+    // is_erc20
+    token->is_erc20 = false;
+
+    // The contract dict has the standard information
+    // so we skip if it's not there.
+    auto* contract = nft->FindDict("contract");
+    if (!contract) {
+      continue;
+    }
+    auto* type = contract->FindString("type");
+    if (!type) {
+      continue;
+    }
+
+    // is_erc721
+    if (coin == mojom::CoinType::ETH) {
+      bool is_erc721 = base::EqualsCaseInsensitiveASCII(*type, "ERC721");
+      if (!is_erc721) {
+        continue;
+      }
+      token->is_erc721 = true;
+    } else {  // mojom::CoinType::SOL
```

Comment:
@yrliou is the new FVM going to be CoinType::ETH or CoinType::FIL? If it's FIL then this would branch in an odd way with that chain.

Reply:
@kdenhartog it's ETH.